packages/gatsby/cache-dir/ssr-develop-static-entry.js

```diff
@@ -111,7 +115,66 @@ export default (pagePath, isClientOnlyPage, callback) => {
 
     const pageData = getPageData(pagePath)
 
-    const componentChunkName = pageData?.componentChunkName
+    const { componentChunkName, staticQueryHashes = [] } = pageData
+
+    let scriptsAndStyles = flatten(
+      [`commons`].map(chunkKey => {
+        const fetchKey = `assetsByChunkName[${chunkKey}]`
+
+        let chunks = get(stats, fetchKey)
+        const namedChunkGroups = get(stats, `namedChunkGroups`)
+
+        if (!chunks) {
+          return null
+        }
+
+        chunks = chunks.map(chunk => {
+          if (chunk === `/`) {
+            return null
+          }
+          return { rel: `preload`, name: chunk }
+        })
+
+        namedChunkGroups[chunkKey].assets.forEach(asset =>
+          chunks.push({ rel: `preload`, name: asset })
+        )
+
+        const childAssets = namedChunkGroups[chunkKey].childAssets
+        for (const rel in childAssets) {
+          chunks = concat(
+            chunks,
+            childAssets[rel].map(chunk => {
+              return { rel, name: chunk }
+            })
+          )
+        }
+
+        return chunks
+      })
+    )
+      .filter(s => isObject(s))
+      .sort((s1, s2) => (s1.rel == `preload` ? -1 : 1)) // given priority to preload
+
+    scriptsAndStyles = uniqBy(scriptsAndStyles, item => item.name)
```

Comment:
If `scriptsAndStyles` was a `Map` to begin with, it wouldn't need to do that relatively expensive step here. And it can still return an array (`[...scriptsAndStyles.values()]`).